### ISSUE TITLE
Numerical overflow in StdScaler

### ISSUE THREAD

**Issue by shchur** [awslabs/gluon-ts#3070]
## Description
StdScaler is prone to numerical overflow because of the squaring operation [here](https://github.com/awslabs/gluonts/blob/dev/src/gluonts/torch/scaler.py#L172).

## To Reproduce

```python
import torch
from gluonts.torch.scaler import StdScaler

x = torch.empty(5, 10).normal_(mean=0, std=1e20)
weights = torch.ones_like(x)
scaler = StdScaler(keepdim=True)
x, loc, scale = scaler(x, weights)  # scale contains inf values
```

This leads to models that use StdScaler encountering NaN values, e.g.,
```python
import pandas as pd
from gluonts.torch.model.tft import TemporalFusionTransformerEstimator

data = [{"target": [1e20 * i for i in range(50)], "start": pd.Period("2020-01-01", "D")}]
estimator = TemporalFusionTransformerEstimator(
    prediction_length=30, freq="D", trainer_kwargs={"max_epochs": 1, "accelerator": "cpu"}
)
predictor = estimator.train(data)  # all weights are NaN, forecasts are also all NaN
```
### What doesn't work
- Using `torch.masked.masked_tensor` to hide values with 0 weight

## Environment
- Operating system: Linux
- Python version: 3.10
- GluonTS version: 0.14.2

(Add as much information about your environment as possible, e.g. dependencies versions.)

**Comment by lostella** [comment]
@shchur do you think you can take a stab at this?
